linker/shared/src/main/scala/org/scalajs/linker/checker/IRChecker.scala

```diff
@@ -522,21 +499,18 @@ private final class IRChecker(unit: LinkingUnit, logger: Logger) {
     }
   }
 
-  private def checkExportedPropertyName(pName: PropertyName,
+  private def checkExportedPropertyName(propName: Tree,
       classDef: LinkedClass, isTopLevel: Boolean)(
       implicit ctx: ErrorContext): Unit = {
-    pName match {
-      case _: Ident =>
-        throw new AssertionError("Exported method may not have Ident as name")
-
+    propName match {
       case StringLiteral(name) =>
-        if (name.contains("__"))
+        if (!classDef.kind.isJSClass && name.contains("__"))
```

Comment:
Yes and no. `__` had to be ruled out in JS classes because they could clash with the logical names of `ComputedName`s. Since logical names, the restriction is lifted.

It does fix a user-visible issue, though. I filed #3784 for that, and I added a test. But that's a by-product; it's not independent.